Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLReader.java

```diff
@@ -0,0 +1,561 @@
+     *
+     * @param root
+     *            root node of the XML DOM tree
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @return the root node of a linked data tree
+     */
+    private static Node toNode(Element root, String documentNS, Storage storage) {
+        return parseNodeElement(root, "", Space.DEFAULT, documentNS, true, storage);
+    }
+
+    /**
+     * Creates a Node from a file.
+     *
+     * @param path
+     *            file to read
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the linked data node
+     * @throws SAXException
+     *             if the file is not wellformed
+     * @throws IOException
+     *             if the reading fails
+     */
+    public static Node toNode(File path, Storage storage) throws SAXException, IOException {
+        return toNode(parseXML(new FileInputStream(path), Optional.empty()), uriForFile(path), storage);
+    }
+
+    /**
+     * Creates a Node from an arbitrary input resource.
+     *
+     * @param bytes
+     *            input data
+     * @param encoding
+     *            character encoding to use
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the linked data node
+     * @throws SAXException
+     *             if the file is not well-formed
+     * @throws IOException
+     *             if the reading fails
+     */
+    public static Node toNode(InputStream bytes, Optional<String> encoding, String documentNS, Storage storage)
+            throws SAXException, IOException {
+        return toNode(parseXML(bytes, encoding), documentNS, storage);
+    }
+
+    /**
+     * Create a Node from an XML String.
+     *
+     * @param data
+     *            String to parse
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the created Node
+     * @throws SAXException
+     *             if the XML is semantically wrong
+     */
+    public static Node toNode(String data, String documentNS, Storage storage) throws SAXException {
+        try {
+            return toNode(new ByteArrayInputStream(data.getBytes("UTF-16")), Optional.of("UTF-16"), documentNS,
+                storage);
+        } catch (UnsupportedEncodingException e) {
+            /* "UTF-16" is always supported by Java */
+            throw new IllegalStateException(e.getMessage(), e);
+        } catch (IOException e) {
+            /* there is no IOException to expect when reading from a String */
```

Comment:
Why you are catching an exception which should not be thrown as you mentioned this in your comment?

Reply:
Because otherwise, the static method `toNode(String, String, Storage)` would have to declare `throws IOException` which it never does. Which would mean any usage of the method would have to care for that exception that is never thrown.

Reply:
Your method already throws a `SAXException` in which case the programmer must react. Throwing a `RuntimeException` instead of a declared `IOException` is not a good way. In most cases `RuntimeException` are not fetched and broke the whole application in a not user friendly way.

Reply:
If compiler informs that IOException can be thrown it should be declared in throws.

Reply:
A `SAXException` *can* happen, because the parsing of the XML may fail (like for `"<foo>bar</a>"`, which is not well-formed). The `IOException` *cannot* happen, because reading from a string does not cause a device error (such as: no read permission, network not responding, defective hard disk, …). It is only declared by the interface because you can pass various readers to the interface (also readers for ‘normal’ files), in which case this exception could happen.

For why I handle this situation like this, see [these recommendations](http://www.odi.ch/prog/design/newbies.php#25).

Reply:
Maybe for developer it is useful to make such a constructs, but for production not. RuntimeException crashes the whole application and is not user friendly. 

Additionally our coding guideline says that exceptions should be thrown up to form classes where they can be presented to the user in the friendly way as a message, not as the crash of the whole application.

In my personal opinion it is just adding not necessary blur to the implementation.

Reply:
The method will not crash, because there are no IOException when reading from a string. A method should not declare an exception that it never throws.